### ISSUE TITLE
The extension failed to load due to a critical error. Please enter Debug Mode to understand the cause

### ISSUE THREAD

**Issue by putrasukardi** [facebookincubator/facebook-for-magento#107]
i successfully installed the plug in, but when I try to load Facebook Ads Extension, I got the following error:
The extension failed to load due to a critical error. Please enter Debug Mode to understand the cause

proceed to enter debug mode, but debug test indicates "No Issues Detected in Test"
![fb ad extension1](https://user-images.githubusercontent.com/2943943/50255208-81d38400-0423-11e9-8eae-80c9eacd62c6.png)
![fb ad extension](https://user-images.githubusercontent.com/2943943/50255209-826c1a80-0423-11e9-98a0-1f3fcbd91984.jpg)

**Comment by tosswithim** [comment]
We are facing same issues with Error discovered when fetching product samples.
`"<br />\n<b>Fatal error</b>: Call to undefined function mb_strlen() in <b>/home/patoys/public_html/includes/src/Facebook_AdsExtension_Model_FBProductFeed.php</b> on line <b>115</b><br />\n"`

What can we Do?

**Comment by tosswithim** [comment]
> 
> 
> We are facing same issues in Magento 1.9 with Error discovered when fetching product samples.
> `"<br />\n<b>Fatal error</b>: Call to undefined function mb_strlen() in <b>/home/patoys/public_html/includes/src/Facebook_AdsExtension_Model_FBProductFeed.php</b> on line <b>115</b><br />\n"`
> 
> What can we Do?

fix it after enabled "mbstring" from WHM

**Comment by adrianluican** [comment]
Hi, 

I'm facing the same issue but i don't understand the solution. What exactly i should do? 

Regards,
Adrian

**Comment by adrianluican** [comment]
> > Hi,
> > I'm facing the same issue but i don't understand the solution. What exactly i should do?
> > Regards,
> > Adrian
> 
> enabled "mbstring" from WHM

Hello, 

mbstring is enabled but it still doesn't work. 

mbstring	enabled Multibyte string engine	libmbfl HTTP input encoding translation	disabled libmbfl version	1.3.2 mbstring extension makes use of "streamable kanji code filter and converter", which is distributed under the GNU Lesser General Public License version 2.1.

**Comment by adrianluican** [comment]
Hi, 

Any update on this. I'm still stuck with this error. Please give me support on this.

**Comment by NewprintCanada** [comment]
We're having the same issue here. Any update?

I found these in the var log in regards to the facebook extension:

2019-03-12T23:56:45+00:00 INFO (6): ERROR2:vsprintf(): Too few arguments @ /var/www/html/app/code/core/Mage/Core/Model/Translate.php L416

[facebook_product_feed_exception.log](https://github.com/facebookincubator/facebook-for-magento/files/2959196/facebook_product_feed_exception.log)

**Comment by eBusinessAtlantique** [comment]
Hello
Same problem as @NewprintCanada 
2019-03-15T18:22:45+00:00 INFO (6): ERROR2:vsprintf(): Too few arguments @ /home/xxx/domains/xxxx/public_html/app/code/core/Mage/Core/Model/Translate.php L416

Admin works fine but I have error 500 on front end.

No error in PHP log.

Where does the error come from ? Any hint appreciated.

Thanks

**Comment by A-Kits** [comment]
> We're having the same issue here. Any update?
> 
> I found these in the var log in regards to the facebook extension:
> 
> 2019-03-12T23:56:45+00:00 INFO (6): ERROR2:vsprintf(): Too few arguments @ /var/www/html/app/code/core/Mage/Core/Model/Translate.php L416
> 
> [facebook_product_feed_exception.log](https://github.com/facebookincubator/facebook-for-magento/files/2959196/facebook_product_feed_exception.log)

I have same issue, Any idea to fix it ?

**Comment by viable-hartman** [comment]
> Hello
> Same problem as @NewprintCanada
> 2019-03-15T18:22:45+00:00 INFO (6): ERROR2:vsprintf(): Too few arguments @ /home/xxx/domains/xxxx/public_html/app/code/core/Mage/Core/Model/Translate.php L416
> 
> Admin works fine but I have error 500 on front end.
> 
> No error in PHP log.
> 
> Where does the error come from ? Any hint appreciated.
> 
> Thanks

I don't believe the vsprintf error is related to the 500 error on the frontend you all are experiencing, I believe it is related to kinda dumb translate function code in Magento Core.  Mage Core uses the controversial PHP **Error Control Operator** to ignore substitution errors when calling vsprintf.  I believe vsprintf only errors when the args array is empty however, so something like this should fix the vsprintf warning and make the Mage code a bit less controversial:

        --- a/app/code/core/Mage/Core/Model/Translate.php
        +++ b/app/code/core/Mage/Core/Model/Translate.php
        @@ -412,10 +412,10 @@ class Mage_Core_Model_Translate

        -
        -        $result = @vsprintf($translated, $args);
        -        if ($result === false) {
        -            $result = $translated;
        -         }
        +        $result = (empty($args)) ? $translated : vsprintf($translated, $args);

Now, the 500 error on the frontend is likely because you are on an Operating System that honors _CASE_ and your PHP is attempting to load modules with their case sensitive class names.  If you look in your repo at **app/code/community/Facebook/AdsExtension/Block/common.php** you'll probably notice that it has a file name of **common.php** all lower case.  Since Magento 1 still auto-loads classes the old PHP way, this file needs to be **Common.php** not **common.php** for your case sensitive OS and PHP to load it properly.  This file is extended by other blocks, so even those blocks will fail when this file name is lowercase.

How this file got named this way in their release, I don't know, but correcting it fixed the errors for me, and I see they have corrected it in the _master_ branch, so installing from that would also resolve this issue.

**Comment by ueltonVeslei** [comment]
> > Hello
> > Same problem as @NewprintCanada
> > 2019-03-15T18:22:45+00:00 INFO (6): ERROR2:vsprintf(): Too few arguments @ /home/xxx/domains/xxxx/public_html/app/code/core/Mage/Core/Model/Translate.php L416
> > Admin works fine but I have error 500 on front end.
> > No error in PHP log.
> > Where does the error come from ? Any hint appreciated.
> > Thanks
> 
> I don't believe the vsprintf error is related to the 500 error on the frontend you all are experiencing, I believe it is related to kinda dumb translate function code in Magento Core. Mage Core uses the controversial PHP **Error Control Operator** to ignore substitution errors when calling vsprintf. I believe vsprintf only errors when the args array is empty however, so something like this should fix the vsprintf warning and make the Mage code a bit less controversial:
> 
> ```
>     --- a/app/code/core/Mage/Core/Model/Translate.php
>     +++ b/app/code/core/Mage/Core/Model/Translate.php
>     @@ -412,10 +412,10 @@ class Mage_Core_Model_Translate
> 
>     -
>     -        $result = @vsprintf($translated, $args);
>     -        if ($result === false) {
>     -            $result = $translated;
>     -         }
>     +        $result = (empty($args)) ? $translated : vsprintf($translated, $args);
> ```
> 
> Now, the 500 error on the frontend is likely because you are on an Operating System that honors _CASE_ and your PHP is attempting to load modules with their case sensitive class names. If you look in your repo at **app/code/community/Facebook/AdsExtension/Block/common.php** you'll probably notice that it has a file name of **common.php** all lower case. Since Magento 1 still auto-loads classes the old PHP way, this file needs to be **Common.php** not **common.php** for your case sensitive OS and PHP to load it properly. This file is extended by other blocks, so even those blocks will fail when this file name is lowercase.
> 
> How this file got named this way in their release, I don't know, but correcting it fixed the errors for me, and I see they have corrected it in the _master_ branch, so installing from that would also resolve this issue.

here a have this issue with git, and solved with this command line:

git mv -f app/code/community/Facebook/AdsExtension/Block/common.php app/code/community/Facebook/AdsExtension/Block/Common.php
